libs/core/langchain_core/indexes/base.py

```diff
@@ -0,0 +1,369 @@
+
+        Returns:
+           A dict ``{"succeeded": [...], "failed": [...]}`` with the IDs of the
+           documents that were successfully deleted and the ones that failed to be
+           deleted.
+
+        Raises:
+            ValueError: if ids are not provided.
+        """
+        if ids is None:
+            raise ValueError("Must provide ids to delete.")
+        if filters:
+            kwargs = {"filters": filters, **kwargs}
+        if kwargs:
+            warnings.warn(
+                "Only deletion by ids is supported for this integration, all other "
+                f"arguments are ignored. Received {kwargs=}"
+            )
+        return await self.adelete_by_ids(ids)
+
+    def lazy_get(
+        self,
+        *,
+        ids: Optional[Iterable[str]] = None,
+        filters: Union[
+            StructuredQuery, Dict[str, Any], List[Dict[str, Any]], None
+        ] = None,
+        **kwargs: Any,
+    ) -> Iterable[Document]:
+        """Default implementation only supports get by id.
+
+        Override this method if the integration supports get by other parameters.
+
+        Args:
+            ids: IDs of the documents to get. Must be specified.
+            **kwargs: Other keywords args not supported by default. Will be ignored.
+
+        Yields:
+           Document.
+
+        Raises:
+            ValueError: if ids are not provided.
+        """
+        if ids is None:
+            raise ValueError("Must provide ids to get.")
+        if filters:
+            kwargs = {"filters": filters, **kwargs}
+        if kwargs:
+            warnings.warn(
+                "Only deletion by ids is supported for this integration, all other "
+                f"arguments are ignored. Received {kwargs=}"
+            )
+        return self.lazy_get_by_ids(ids)
+
+    async def alazy_get(
+        self,
+        *,
+        ids: Optional[AsyncIterable[str]] = None,
+        filters: Union[
+            StructuredQuery, Dict[str, Any], List[Dict[str, Any]], None
+        ] = None,
+        **kwargs: Any,
+    ) -> AsyncIterable[Document]:
+        """Default implementation only supports get by id.
+
+        Override this method if the integration supports get by other parameters.
+
+        Args:
+            ids: IDs of the documents to get. Must be specified.
+            **kwargs: Other keywords args not supported by default. Will be ignored.
+
+        Yields:
+           Document.
+
+        Raises:
+            ValueError: if ids are not provided.
+        """
+        if ids is None:
+            raise ValueError("Must provide ids to get.")
+        if filters:
+            kwargs = {"filters": filters, **kwargs}
+        if kwargs:
+            warnings.warn(
+                "Only deletion by ids is supported for this integration, all other "
+                f"arguments are ignored. Received {kwargs=}"
+            )
+        return await self.alazy_get_by_ids(ids)
+
```

Comment:
Considering breaking into `get_by_id` and `get_by_filter`. `get_by_id` is a subset of `get_by_filter`. Use two distinct ones for ease of implementation and testing.

Later we could use `$ID` or `$.ID` to denote special cased fields in filters (if we ever cared)

---

There's a more general get by query that could support sorting by metadata fields (if we really cared to add that). It's generally useful for working with documents to sort by publication datetime. Seems like an unnecessary complication right now and not part of a basic vector index